homeassistant/components/http.py

```diff
@@ -45,6 +51,63 @@
 }, extra=vol.ALLOW_EXTRA)
 
 
+def collect_hash(config):
+    """
+    find the password/hash information in http configuration.
+
+    Args:
+        config: configuration['http']
+
+    Returns:
+        '$<algorithm>$<salt>$<hash>'
+        or
+        None if password or hash are not configured
+    """
+    if CONF_API_HASH in config:
+        hash_ = config[CONF_API_HASH]
+    elif CONF_API_PASSWORD in config:
+        hash_ = new_hash(config[CONF_API_PASSWORD])
+    else:
+        return None
+    return split_hash(hash_)
+
+
+def new_hash(password, salt=None):
+    """
+    create new sha512 hash from password and random salt.
+
+    Args:
+        password: password string
+        salt: arbitrary string
+
+    Returns: '$<algorithm>$<salt>$<hash>'
+
+    """
+    if not salt:
+        alpha_numeric = string.ascii_letters + string.digits
+        salt = ''.join(random.sample(alpha_numeric, SALT_LENGTH))
+    assert '$' not in salt, 'salt contains "$"'
+    hasher = hashlib.sha512()
```

Comment:
I'm sorry, I'm not sure I understand the context of your question.

What is driving the requirement to use the crypt module, what is driving the requirement to encrypt passwords at all, what is driving the requirement to encrypt password in this way, or something else that I haven't considered?